vm/src/codecs.rs

```diff
@@ -428,3 +440,255 @@ fn backslashreplace_errors(err: PyObjectRef, vm: &VirtualMachine) -> PyResult<(S
     }
     Ok((out, range.end))
 }
+
+fn namereplace_errors(err: PyObjectRef, vm: &VirtualMachine) -> PyResult<(String, usize)> {
+    if err.isinstance(&vm.ctx.exceptions.unicode_encode_error) {
```

Comment:
nitpick: I'd handle the else clause first and return the err and, in effect, dedent this if.

Reply:
I was just patterning the general structure from [the CPython code](https://github.com/python/cpython/blob/dd7b816ac87e468e2fa65ce83c2a03fe1da8503e/Python/codecs.c#L959-L1051). Perhaps that improvement could be upstreamed, especially if it boosts performance.